pkg/provider/azure_loadbalancer.go

```diff
@@ -858,22 +870,30 @@ func (az *Cloud) getServiceLoadBalancerStatus(service *v1.Service, lb *network.L
 }
 
 // pips: a non-nil pointer to a slice of existing PIPs, if the slice being pointed to is nil, listPIP would be called when needed and the slice would be filled
-func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service, pips *[]network.PublicIPAddress) (string, bool, error) {
-	if name, found := service.Annotations[consts.ServiceAnnotationPIPNameDualStack[false]]; found && name != "" {
+func (az *Cloud) determinePublicIPName(clusterName string, service *v1.Service, pips *[]network.PublicIPAddress, isIPv6 bool) (string, bool, error) {
+	if name := getServicePIPName(service, isIPv6); name != "" {
 		return name, true, nil
 	}
-	isIPv6 := utilnet.IsIPv6String(service.Spec.ClusterIP)
-	if ipPrefix, ok := service.Annotations[consts.ServiceAnnotationPIPPrefixIDDualStack[false]]; ok && ipPrefix != "" {
-		return az.getPublicIPName(clusterName, service, isIPv6), false, nil
-	}
 
 	pipResourceGroup := az.getPublicIPAddressResourceGroup(service)
+	if id := getServicePIPPrefixID(service, isIPv6); id != "" {
+		pipName, err := az.getPublicIPName(clusterName, service, pipResourceGroup, pips, isIPv6)
+		if err != nil {
+			return "", false, err
+		}
+		return pipName, false, nil
```

Comment:
Can we return pipName, false, and err here to reduce complexity?

Reply:
Good idea. Updated.